eureka/S3_data_reduction/background.py

```diff
@@ -62,20 +64,21 @@ def BGsubtraction(data, meta, log, isplots):
     # Write background
     def writeBG(arg):
         bg_data, bg_mask, n = arg
-        data.subbg[n] = bg_data
-        data.submask[n] = bg_mask
+        data['bg'][n] = bg_data
```

Comment:
I wonder if it is possible to write xarrays using `data.bg = bg_data` by writing a custom __setattr__ function for the Astraeus objects. It'd save a lot of annoyance and inconsistency between accessing and setting attributes. I don't know why that isn't already possible (maybe there's a good reason they forbid it), but it seems to me like a simple setattr function that converts the `data.bg = bg_data` syntax to the currently required `data['bg'] = bg_data` syntax (programatically so it's not just for one particular attribute) should be pretty easy. I can try fiddling around with that if you're interested.

Reply:
Yeah, let me know if you come up with something. I'll look into it as well after we get v0.1 out the door.

Reply:
I came up with something that works and confirmed that it doesn't cause any unusual behaviours as far as I could think to test (but I'm very much an xarray beginner). Submitted a PR on Astraeus so we can discuss this more there